RCTAppleHealthKit/RCTAppleHealthKit+Methods_ClinicalRecords.m

```diff
@@ -0,0 +1,102 @@
+//
+//  RCTAppleHealthKit+Methods_ClinicalRecords.m
+//  RCTAppleHealthKit
+//
+//  This source code is licensed under the MIT-style license found in the
+//  LICENSE file in the root directory of this source tree.
+
+#import "RCTAppleHealthKit+Methods_ClinicalRecords.h"
+#import "RCTAppleHealthKit+Queries.h"
+#import "RCTAppleHealthKit+Utils.h"
+
+@implementation RCTAppleHealthKit (Methods_ClinicalRecords)
+
+- (void)clinicalRecords_getClinicalRecords:(NSDictionary *)input callback:(RCTResponseSenderBlock)callback
+{
+    NSString *type = [RCTAppleHealthKit stringFromOptions:input key:@"type" withDefault:nil];
+    if(type == nil){
+     callback(@[RCTMakeError(@"type is required in options", nil, nil)]);
+     return;
+    }
+    
+    [self clinicalRecords_getClinicalRecordsOfType:type input:input callback:callback];
+}
+
+- (void)clinicalRecords_getClinicalRecordsOfType:(NSString *)type input:(NSDictionary *)input callback:(RCTResponseSenderBlock)callback
+{
+    HKClinicalType *recordType = [RCTAppleHealthKit clinicalTypeFromName:type];
+    if (recordType == nil) {
+        callback(@[RCTMakeError(@"invalid type", nil, nil)]);
+        return;
+    }
+    
+    NSUInteger limit = [RCTAppleHealthKit uintFromOptions:input key:@"limit" withDefault:HKObjectQueryNoLimit];
+    BOOL ascending = [RCTAppleHealthKit boolFromOptions:input key:@"ascending" withDefault:false];
+    
+    NSDate *startDate = [RCTAppleHealthKit dateFromOptions:input key:@"startDate" withDefault:nil];
+    if(startDate == nil){
+     callback(@[RCTMakeError(@"startDate is required in options", nil, nil)]);
+     return;
+    }
+    
+    NSDate *endDate = [RCTAppleHealthKit dateFromOptions:input key:@"endDate" withDefault:[NSDate date]];
+    NSPredicate * predicate = [RCTAppleHealthKit predicateForSamplesBetweenDates:startDate endDate:endDate];
+    
+    [self fetchClinicalRecordsOfType:recordType predicate:predicate ascending:ascending limit:limit completion:^(NSArray *results, NSError *error) {
+        if(results){
+            callback(@[[NSNull null], results]);
+            return;
+        } else {
+            callback(@[RCTJSErrorFromNSError(error)]);
+            return;
+        }
+    }];
+}
+
+- (void)clinicalRecords_getAllergyRecords:(NSDictionary *)input callback:(RCTResponseSenderBlock)callback
+{
+    [self clinicalRecords_getClinicalRecordsOfType:@"AllergyRecord" input:input callback:callback];
```

Comment:
what is the difference between calling these methods (lines 56-91) and calling `clinicalRecords_getClinicalRecords` with the type in the input?

Reply:
I definitely don't _disagree_ with you. In fact removing the specific getters (e.g. getAllergyRecords) will greatly reduce the size of this pull request because then I can delete a bunch of documentation :)

I added both the methods in (56-91) and `clinicalRecords_getClinicalRecords` to give the clinical records a similar API footprint to the fitness methods. At first glance I thought that calling for example `getSamples` with a type of `StepCount` was is equivalent to calling `getStepCount`, though on closer inspection they do return different results.

I'm definitely on board with simplifying the api to just the `getClinicalRecords` method